meetings/2018-01-17/minutes.md

```diff
@@ -0,0 +1,417 @@
+the discussion to ensure that everyone involved is brought to a common
+understanding of the existing tooling ecosystem.
+
+### Scalameta & SemanticDB - Eugene Burmako, Twitter
+
+Scalameta has almost 50 open-source developers, and as been funded by EPFL,
+JetBrains, Scala Center and Twitter. Scalameta provides APIs optimised for
+tools, designed to be cross-platform, and to work with multiple compilers. It
+provides a syntactic API and a semantic API. SemanticDB is an interchange
+format for Scalameta based on protocol buffers. A SemanticDB document
+corresponds to a source file. SemanticDB can be generated with a compiler
+plugin. The file contains full details of symbols, and details of how names and
+positions and resolve to symbols.
+
+There's a SemanticDB plugin for Scalac 2.11 and 2.12. Many open-source Scala
+projects have been indexed already, including all of Twitter's code. It enables
+Scalafix, Metadoc, Metals and several internal tools at Twitter (which Eugene
+is not at liberty to talk about). It does not require a running compiler, and
+is very fast, even for large databases. The information can be put into an SQL
+database. As an example, two million lines of code can be indexed into an 800MB
+SQLite database, and offers 10ms access times.
+
+Eugene shared the [slides from his presentation](http://goo.gl/TsFfqC).
+
+### Metals - Eugene Burmako, Twitter
+
+Metals ("scalaMETA Language Server") in an LSP implementation based on
+Scalameta, developed primarily by Ólafur Páll Geirsson. Olaf was not able to
+attend the full meeting, so Eugene Burmako presented on his behalf.
+Unfortunately, this part of the presentation was not fully captured on the
+recording, but Olaf shared a document showing some of the [features provided by
+Metals](https://geirsson.com/assets/metals/).
+
+### Bloop - Jorge Vicente Cantero, Scala Center
+
+Bloop is a build server, integrating with all build tools including sbt, Maven
+and Gradle, which can be considered an small enhancement over the existing
+Scalac compiler. It was created by Jorge and Martin Duhem. A stable release
+will be available soon.
+
+Jorge also shared a document detailing a draft proposal for a [Build Server
+Protocol](https://github.com/scalacenter/bsp/blob/master/docs/bsp.md) or "BSP".
+
+### Sbt LSP implementation - Eugene Yokata, Lightbend
+
+The sbt LSP server predates Eugene Yokata starting at Lightbend: there has been
+an sbt server since 2015, with sepration of front-end and backend, running on
+two JVMs. A simpler approach was proposed in 2016, running on a single JVM. The
+server can be thought of an event-based CQRS system, where commands arrive, and
+responses are asynchronously returned. Sbt has full knowledge of the build,
+including library management and test integration. Sbt can provide richer
+features than LSP, including information like packaging and deployment. Error
+messages can be sent as JSON to clients.
```

Comment:
The point I wanted to make is that while sbt as build tool has full knowledge of the build (source, Zinc, libraries, tests, extensible plugins) and act as textbased-IDE, there's a growing demand for richer development environment (jumping to error position, code navigation, tab completion, visual debugging etc). I see LSP to be an intersection of these two, and the feature that's relevant in sbt 1 is called Event Logging.

![sbt_server__lsp_discussion__2018_jan_](https://user-images.githubusercontent.com/184683/35349266-7a9ec43e-0108-11e8-816f-603915d72ab0.png)

Here's the slidedeck I prepared for the meeting (but wasn't able to show beyond the first page) - https://www.slideshare.net/EugeneYokota/sbt-server-lsp-discussion-2018-jan